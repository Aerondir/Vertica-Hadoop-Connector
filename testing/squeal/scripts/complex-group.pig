fs -copyFromLocal $script/url.log compgroup/url.log
A = LOAD 'compgroup/url.log' USING PigStorage('|') AS (url:chararray,outlink:chararray);
B = GROUP A BY url;
X = FOREACH B {
        FA = FILTER A BY outlink == 'www.xyz.org';
        PA = FA.outlink;
        DA = DISTINCT PA;
	FA2 = FILTER A BY outlink == 'www.cvn.org';
	PA2 = FA2.outlink;
        GENERATE group, COUNT(DA), COUNT(PA2);
};
STORE X INTO 'compgroup/output';
cat compgroup/output;
fs -rmr compgroup;
/*

# select url,count(*), count(distinct outlink) from B group by url where outlink == 'www.xyz.org';
#
# 1 1
# 1 -1
C = FOREACH B GENERATE 1, group, A;

A = LOAD 'data' AS (f1:int,f2:int);
B = GROUP A BY f1;
X = FOREACH B {
  PA = A.f2;
  DA = DISTINCT PA;
  GENERATE group, SUM(DA.f2*DA.f2);
};
*/