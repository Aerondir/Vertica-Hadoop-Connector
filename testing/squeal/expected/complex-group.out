10/10/18 23:56:44 INFO squeal.Squeal: Logging error messages to: /home/rvenkatesh/dev/Linux64/Test/squeal_smoke/pigshell.out
Constructing server
Configuring vertica connection
Translating...
Running script schema refresh(SQL)
ROLLBACK: schema "squeal_PigLatin_complex_group_pig" does not exist
Time: 57ms
Running script sql setup(SQL)
Time: 56ms
Running script test sql(SQL)
Time: 59ms
Running script load3(PIG)
Use GenericOptionsParser for parsing the arguments. Applications should implement Tool for the same.
Total input paths to process : 1
Time: 19111ms
Running script populate vdata4(SQL)
Time: 334ms
Running script store vdata5(PIG)
Use GenericOptionsParser for parsing the arguments. Applications should implement Tool for the same.
Time: 17779ms
Total time: 37449ms
www.aaa.com	0	1
www.ccc.com	0	0
www.ddd.com	1	0
www.www.com	1	0
Deleted hdfs://hermetic:9935/user/rvenkatesh/compgroup
